lib/source/Instruction.cpp

```diff
@@ -1614,6 +1633,42 @@ spvgentwo::Instruction* spvgentwo::Instruction::toSpecOp()
 	return this;
 }
 
+spvgentwo::Instruction* spvgentwo::Instruction::opPtrEqual(Instruction* _pLeftPtr,
+	Instruction* _pRightPtr)
+{
+	auto module = getModule();
+
+	if (_pLeftPtr->getType()->isPointer() && _pRightPtr->getType()->isPointer()
+		&& _pLeftPtr->getType() == _pRightPtr->getType())
+	{
+		auto t = module->addType(module->newType().Bool());
```

Comment:
consider using module->type<bool>(); here

please follow the guide for implementing new instructions: https://github.com/rAzoR8/SpvGenTwo/blob/master/DOCUMENTATION.md#Implementing-New-Instructions and also update inferResultType in https://github.com/rAzoR8/SpvGenTwo/blob/master/lib/source/TypeInferenceAndValiation.cpp accordingly